src/app/qgscrashreport.h

```diff
@@ -0,0 +1,114 @@
+/***************************************************************************
+  qgscrashreport.h - QgsCrashReport
+
+ ---------------------
+ begin                : 16.4.2017
+ copyright            : (C) 2017 by Nathan Woodrow
+ email                : woodrow.nathan@gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSCRASHREPORT_H
+#define QGSCRASHREPORT_H
+
+#include "qgis_app.h"
+
+#include <QObject>
+
+
+/**
+ * Include information to generate user friendly crash report for QGIS.
+ */
+class APP_EXPORT QgsCrashReport
+{
+  public:
+
+    /**
+     * Represents a line from a stack trace.
+     */
+    struct StackLine
+    {
+      QString ModuleName;
+      QString SymbolName;
+      QString FileName;
+      QString LineNumber;
+
+      /**
+       * Check if this stack line is part of QGIS.
+       * @return True if part of QGIS.
+       */
+      bool IsQGISModule();
+
+      /**
+       * Check if this stack line is valid.  Considered valid when the filename and line
+       * number are known.
+       * @return True of the line is valid.
+       */
+      bool IsValid();
+    };
+
+    /**
+     * Include information to generate user friendly crash report for QGIS.
+     */
+    QgsCrashReport();
+
+  public:
+    enum Flag
+    {
+      Stack                = 1,
+      Plugins              = 2,
+      ProjectDetails       = 4,
+      SystemInfo           = 6,
+      QGISInfo             = 8,
+    };
+    Q_DECLARE_FLAGS( Flags, Flag )
+
+    /**
+     * Sets the stack trace for the crash report.
+     * @param value A string list for each line in the stack trace.
+     */
+    void setStackTrace( const QList<QgsCrashReport::StackLine> value ) { mStackTrace = value; }
+
+    /**
+     * Returns the stack trace for this report.
+     * @return A string list for each line in the stack trace.
+     */
+    const QList<QgsCrashReport::StackLine> StackTrace() { return mStackTrace; }
```

Comment:
Can you change this to:

    QList<QgsCrashReport::StackLine> StackTrace() const 

(ie const at end not start)